ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_filter.proto

```diff
@@ -42,26 +42,38 @@ message Filters {
 // the ``template_ids`` or that match one of the ``interface_filters``.
 message InclusiveFilters {
 
-  // A collection of templates for which the payload will be included in the
+  // A collection of templates for which the data will be included in the
   // ``create_arguments`` of a matching ``CreatedEvent``.
   // SHOULD NOT contain duplicates.
-  // All ``template_ids`` needs to be valid: corresponding template should be defined in one of
+  // All ``template_ids`` need to be valid: corresponding templates should be defined in one of
   // the available packages at the time of the query.
+  // Deprecated in favor of the `template_filters`. If the `template_filters` field is set,
+  // the `template_ids` field is ignored.
   // Optional
-  repeated Identifier template_ids = 1;
+  repeated Identifier template_ids = 1 [deprecated = true];
 
   // Include an ``InterfaceView`` for every ``InterfaceFilter`` matching a contract.
   // The ``InterfaceFilter``s MUST use unique ``interface_id``s.
-  // All ``interface_id`` needs to be valid: corresponding interface should be defined in one of
-  // the available packages at the time of the query.
   // Optional
   repeated InterfaceFilter interface_filters = 2;
+
+  // A collection of templates for which the data will be included in the
+  // ``create_arguments`` of a matching ``CreatedEvent``.
+  // SHOULD NOT contain duplicate templates.
+  // In contrast to the ``template_ids`` field, one can specify whether the create event payloads
+  // should be provided alongside the create arguments.
+  // If a contract is simultaneously selected by a template filter and one or more interface filters,
+  // the corresponding ``include_create_event_payload`` are consolidated using an OR operation.
+  // Optional
+  repeated TemplateFilter template_filters = 3;
```

Comment:
@mziolekda : @cocreature just raised the good point that one cannot have a wildcard filter where the create_events cointain the create_event_payload. I bet both PQS and CN will need that.

Reply:
On a slightly related note, we also need this to work for transaction tree streams which we ingest in CN.

Reply:
We are aware of the transaction tree streams issue where currently all templates are in (effectively a wildcard), so the idea is that the filter will only be used to signify what extras to put on the event.
As for the blanket inclusion of the payloads for templates, I am slightly surprised by that. In my opinion ED is a sharp tool that should be used sparsely only for templates that are truly going to be shared off-ledger. I would imagine it would affect a small portion of all templates.

Reply:
> We are aware of the transaction tree streams issue where currently all templates are in (effectively a wildcard), so the idea is that the filter will only be used to signify what extras to put on the event.

Makes sense, that matches my expectation.

> As for the blanket inclusion of the payloads for templates, I am slightly surprised by that. In my opinion ED is a sharp tool that should be used sparsely only for templates that are truly going to be shared off-ledger. I would imagine it would affect a small portion of all templates.

That's partially true. But we are increasingly moving toward PQS being viewed as almost part of the participant to avoid the common objection that it's not part of your participant and also want to include cryptographic evidence in pqs long-term.

By default PQS atm just ingests everything. It makes more sense to me that in that mode it also ingests all blobs.

I don't think this is necessarily blocking (the tree support is blocking for CN) but it seems like a sensible feature to me.